File: Examples/test-suite/perl5/rename_wildcard_runme.pl
use strict;
use warnings;
use Test::More tests => 9;
BEGIN { use_ok('rename_wildcard') }
require_ok 'rename_wildcard';

sub run_ok(&$) {
  my($test, $name) = @_;
  my $tb = Test::More->builder();
  eval { $test->() };
  return $tb->is_eq($@, '', $name);
}

run_ok {
  rename_wildcard::GlobalWildStruct->new()->mm1();
  rename_wildcard::GlobalWildTemplateStructInt->new()->mm1();
  rename_wildcard::SpaceWildStruct->new()->mm1();
  rename_wildcard::SpaceWildTemplateStructInt->new()->mm1();
} 'Wildcard check';
run_ok {
  rename_wildcard::GlobalWildStruct->new()->mm2a();
  rename_wildcard::GlobalWildTemplateStructInt->new()->mm2b();
  rename_wildcard::SpaceWildStruct->new()->mm2c();
  rename_wildcard::SpaceWildTemplateStructInt->new()->mm2d();

  rename_wildcard::GlobalWildTemplateStructInt->new()->tt2b();
  rename_wildcard::SpaceWildTemplateStructInt->new()->tt2d();
}'No declaration';
run_ok {
  rename_wildcard::GlobalWildStruct->new()->mm3a();
  rename_wildcard::GlobalWildTemplateStructInt->new()->mm3b();
  rename_wildcard::SpaceWildStruct->new()->mm3c();
  rename_wildcard::SpaceWildTemplateStructInt->new()->mm3d();

  rename_wildcard::GlobalWildTemplateStructInt->new()->tt3b();
  rename_wildcard::SpaceWildTemplateStructInt->new()->tt3d();
} 'With declaration';
run_ok {
  rename_wildcard::GlobalWildStruct->new()->mm4a();
  rename_wildcard::GlobalWildTemplateStructInt->new()->mm4b();
  rename_wildcard::SpaceWildStruct->new()->mm4c();
  rename_wildcard::SpaceWildTemplateStructInt->new()->mm4d();

  rename_wildcard::GlobalWildTemplateStructInt->new()->tt4b();
  rename_wildcard::SpaceWildTemplateStructInt->new()->tt4d();
} 'Global override too';
run_ok {
  rename_wildcard::GlobalWildStruct->new()->mm5a();
  rename_wildcard::GlobalWildTemplateStructInt->new()->mm5b();
  rename_wildcard::SpaceWildStruct->new()->mm5c();
  rename_wildcard::SpaceWildTemplateStructInt->new()->mm5d();

  rename_wildcard::GlobalWildTemplateStructInt->new()->tt5b();
  rename_wildcard::SpaceWildTemplateStructInt->new()->tt5d();
} '%extend renames';
run_ok {
  rename_wildcard::GlobalWildStruct->new()->opinta();
  rename_wildcard::GlobalWildTemplateStructInt->new()->opintb();
  rename_wildcard::SpaceWildStruct->new()->opintc();
  rename_wildcard::SpaceWildTemplateStructInt->new()->opintd();

  rename_wildcard::GlobalWildTemplateStructInt->new()->opdoubleb();
  rename_wildcard::SpaceWildTemplateStructInt->new()->opdoubled();
} 'operators';
run_ok {
  rename_wildcard::NoChangeStruct->new()->mm1();
  rename_wildcard::NoChangeStruct->new()->mm2();
  rename_wildcard::NoChangeStruct->new()->mm3();
  rename_wildcard::NoChangeStruct->new()->mm4();
  rename_wildcard::NoChangeStruct->new()->mm5();
  rename_wildcard::NoChangeStruct->new()->opint();
  rename_wildcard::SpaceNoChangeStruct->new()->mm1();
  rename_wildcard::SpaceNoChangeStruct->new()->mm2();
  rename_wildcard::SpaceNoChangeStruct->new()->mm3();
  rename_wildcard::SpaceNoChangeStruct->new()->mm4();
  rename_wildcard::SpaceNoChangeStruct->new()->mm5();
  rename_wildcard::SpaceNoChangeStruct->new()->opint();
} 'Wildcard renames expected for these';

File: Examples/test-suite/lua/rename_wildcard_runme.lua
local v=require("rename_wildcard")

-- Wildcard check
v.GlobalWildStruct():mm1()
v.GlobalWildTemplateStructInt():mm1()
v.SpaceWildStruct():mm1()
v.SpaceWildTemplateStructInt():mm1()

-- No declaration
v.GlobalWildStruct():mm2a()
v.GlobalWildTemplateStructInt():mm2b()
v.SpaceWildStruct():mm2c()
v.SpaceWildTemplateStructInt():mm2d()

v.GlobalWildTemplateStructInt():tt2b()
v.SpaceWildTemplateStructInt():tt2d()

-- With declaration
v.GlobalWildStruct():mm3a()
v.GlobalWildTemplateStructInt():mm3b()
v.SpaceWildStruct():mm3c()
v.SpaceWildTemplateStructInt():mm3d()

v.GlobalWildTemplateStructInt():tt3b()
v.SpaceWildTemplateStructInt():tt3d()

-- Global override too
v.GlobalWildStruct():mm4a()
v.GlobalWildTemplateStructInt():mm4b()
v.SpaceWildStruct():mm4c()
v.SpaceWildTemplateStructInt():mm4d()

v.GlobalWildTemplateStructInt():tt4b()
v.SpaceWildTemplateStructInt():tt4d()

-- %extend renames
v.GlobalWildStruct():mm5a()
v.GlobalWildTemplateStructInt():mm5b()
v.SpaceWildStruct():mm5c()
v.SpaceWildTemplateStructInt():mm5d()

v.GlobalWildTemplateStructInt():tt5b()
v.SpaceWildTemplateStructInt():tt5d()

-- operators
v.GlobalWildStruct():opinta()
v.GlobalWildTemplateStructInt():opintb()
v.SpaceWildStruct():opintc()
v.SpaceWildTemplateStructInt():opintd()

v.GlobalWildTemplateStructInt():opdoubleb()
v.SpaceWildTemplateStructInt():opdoubled()

-- Wildcard renames expected for these
v.NoChangeStruct():mm1()
v.NoChangeStruct():mm2()
v.NoChangeStruct():mm3()
v.NoChangeStruct():mm4()
v.NoChangeStruct():mm5()
v.NoChangeStruct():opint()
v.SpaceNoChangeStruct():mm1()
v.SpaceNoChangeStruct():mm2()
v.SpaceNoChangeStruct():mm3()
v.SpaceNoChangeStruct():mm4()
v.SpaceNoChangeStruct():mm5()
v.SpaceNoChangeStruct():opint()
